torch/lib/THD/master_worker/master/generic/THDTensor.cpp

```diff
@@ -635,9 +760,134 @@ void THDTensor_(unfold)(THDTensor *self, THDTensor *src,
   );
 }
 
-// TODO implement
+void THDTensor_(squeeze)(THDTensor *self, THDTensor *src) {
+  int ndim = 0;
+
+  if (!src)
+    src = self;
+
+  THDTensor_(set)(self, src);
+
+  for (std::size_t d = 0; d < src->nDimension; d++) {
+    if (src->size[d] != 1) {
+      if (d != ndim) {
+        self->size[ndim] = src->size[d];
+        self->stride[ndim] = src->stride[d];
+      }
+      ndim++;
+    }
+  }
+
+  /* right now, we do not handle 0-dimension tensors */
+  if(ndim == 0 && src->nDimension > 0) {
+    self->size[0] = 1;
+    self->stride[0] = 1;
+    ndim = 1;
+  }
+  self->nDimension = ndim;
+  masterCommandChannel->sendMessage(
+      packMessage(Functions::tensorSqueeze, self, src),
+      THDState::s_current_worker
+  );
+}
+
+void THDTensor_(squeeze1d)(THDTensor *self, THDTensor *src, int dimension) {
+  if(!src)
+    src = self;
+
+  THArgCheck((dimension >= 0) && (dimension < src->nDimension), 2, "dimension out of range");
+
+  THDTensor_(set)(self, src);
+
+  if (src->size[dimension] == 1 && src->nDimension > 1) {
+    for (std::size_t d = dimension; d < self->nDimension-1; d++) {
+      self->size[d] = self->size[d+1];
+      self->stride[d] = self->stride[d+1];
+    }
+    self->nDimension--;
+  }
+  masterCommandChannel->sendMessage(
+      packMessage(Functions::tensorSqueeze1d, self, src),
+      THDState::s_current_worker
+  );
+}
+
+int THDTensor_(isContiguous)(const THDTensor *self) {
+  long z = 1;
+  for (std::size_t d = self->nDimension - 1; d >= 0; d--) {
```

Comment:
Infinite loop. You must change type since comparison `d >= 0` is always true for `std::size_t` or change loop 'direction' : starting from `0` going up to `self->nDimension - 1`